portfolio/src/main/webapp/script.js

```diff
@@ -12,43 +12,111 @@
-    blogListElement.appendChild(createListElement(blogData.commentsList[i]));
+function addAllPostToBlog(comments) {
+  const blogContainer = document.getElementById('blog-container');
+
+  // comments sorted by most recently posted to resemble a blog
+  for (const commentObject of comments) {
+    blogContainer.append(
+        createPost(commentObject.author, commentObject.comment));
   }
 }
 
-/**
- * Adds a random fact to the page.
- */
-function addRandomfact() {
-  const personalFact = [
-    'Samoyed is my favorite dog breed', 'Horchata is my favorite drink',
-    'I enjoy running', 'Maplestory is my favorite childhood game'
-  ];
+function addPostToBlog() {
+  const blogContainer = document.getElementById('blog-container');
+  const author = emailAddress;
+  const comment = document.getElementById('comment-input').value;
+  blogContainer.prepend(createPost(author, comment));
+}
+
+function createPost(author, comment) {
+  const postContainer = document.createElement('div');
+  postContainer.append(createPostUserImage());
+  postContainer.append(createPostComment(author, comment));
+  return postContainer;
+}
+
+function createPostUserImage() {
+  const userImage = document.createElement('img');
+  userImage.src = 'images/anonymous.png';
+  return userImage;
+}
+
+function createPostComment(author, comment) {
+  const commentContainer = document.createElement('div');
+  commentContainer.className = 'user-comment';
+  addCommentText(commentContainer, 'h3', author);
+  addCommentText(commentContainer, 'p', comment);
+  return commentContainer;
+}
+
+function addCommentText(commentContainer, elementType, content) {
+  const element = document.createElement(elementType);
+  element.innerText = content;
+  commentContainer.append(element);
+}
 
-  // Pick a random fact.
-  const fact = personalFact[Math.floor(Math.random() * personalFact.length)];
+function addComment(clickSubmitEvent) {
+  clickSubmitEvent.preventDefault();
 
-  // Add it to the page.
-  const factContainer = document.getElementById('fact-container');
-  factContainer.innerText = fact;
+  fetch('/comment', {
+    method: 'post',
+    body: new URLSearchParams(
+        new FormData(document.getElementById('comment-form')))
+  })
+      .then(addPostToBlog)
+      .then(resetForm);
 }
+
+function resetForm() {
+  document.getElementById('comment-form').reset();
+}
+
+async function showCommentFormIfLoggedIn() {
+  const response = await fetch('/authentication');
+  const authenticationInfo = await response.json();
+  if (authenticationInfo.isLoggedIn) {
+    resetForm();
+    emailAddress = authenticationInfo.emailAddress;
+  } else {
+    hideCommentForm();
+  }
+}
+
+function hideCommentForm() {
+  document.getElementById('comment-form').style.display = 'none';
+}
+
+// Need global email for addPostToBlog -> to avoid making another fetch request
+// assigned in showCommentFormIfLoggedIn
+// Unsure if this is the right way to go about it though
```

Comment:
Keeping a global variable like this is fine since the application is still pretty simple.  As you continue to add complexity to the app, you'll probably introduce a framework to help manage things, which should also provide a way to handle state like this.